src/console/spcm_control/rx_device.py

```diff
@@ -329,7 +329,7 @@ def _gated_timestamps_stream(self):
                 gate_length = Decimal(str(timestamp_1)) - Decimal(str(timestamp_0))
 
                 # Calculate the number of adc gate sample points (per channel)
-                gate_sample = int(gate_length * (Decimal(str(self.sample_rate)) * Decimal("1e6")))
+                gate_sample = int(round(gate_length * (Decimal(str(self.sample_rate)) * Decimal("1e6"))))
```

Comment:
Why do we need the extra rounding here? We may need to verify that calculated gate_samples matches the predicted ones. Maybe we can perform 100 fast rx and check the logs?

Reply:
I think here it doesn't matter as the `gate_length` (measured by the time stamps) is always a multiple of the sampling rate